app/src/main/java/com/example/countryinfo/CountryDetailsAdapter.kt

```diff
@@ -39,63 +37,67 @@ class CountryDetailsAdapter(
         notifyDataSetChanged()
     }
 
-    class ViewHolder(
-        view: View, private val listener: OnItemClickListener
-    ) : RecyclerView.ViewHolder(view), View.OnClickListener {
-
-        var viewSection: DetailsSections? = null
-        var dialogMessage = emptyList<String>()
-
-        init {
-            itemView.setOnClickListener(this)
-        }
+    class ViewHolder(view: View, private val listener: OnItemClickListener) :
+        RecyclerView.ViewHolder(view) {
 
-        override fun onClick(v: View?) {
-            viewSection?.let { listener.onItemClick(it, dialogMessage) }
-        }
+        private val titles = listOf(itemView.titleText, itemView.titleText2, itemView.titleText3)
 
         interface OnItemClickListener {
-            fun onItemClick(section: DetailsSections, dialogMessage: List<String>)
+            fun onItemClick(state: DetailsViewHolderState)
         }
 
         fun bind(state: DetailsViewHolderState) {
             itemView.labelText.text = state.title
+            itemView.truncateDots.isVisible = false
 
-            with(itemView) {
-                listOf(titleText, titleText2, titleText3).forEachIndexed { index, textView ->
-                    state.info.getOrNull(index)?.let {
-                        textView.apply {
-                            text = it
-                            setBackgroundResource(state.detailsSections.bgShapeId)
-                            isVisible = true
-                        }
-                        if (index == 0) {
-                            viewSection = null
+            titles.forEachIndexed { index, textView ->
+                state.info.getOrNull(index)?.let {
+                    textView.apply {
+                        text = it
+                        if (state.detailsSections.bgShapeId == 0) {
+                            setBackgroundResource(0)
+                            textView.setPadding(0, 0, 0, 0)
                         } else {
-                            viewSection = state.detailsSections
-                            dialogMessage = state.info
-                            itemView.post { checkViewsWidth() }
+                            setBackgroundResource(state.detailsSections.bgShapeId)
                         }
-                    } ?: run { textView.isVisible = false }
+                    }
+                }
+            }
+            if (state.info.size > 1) {
+                itemView.setOnClickListener { listener.onItemClick(state) }
+                checkViewsWidth(itemView)
+            } else {
+                itemView.setOnClickListener(null)
+                for (i in 1 until titles.size){
+                    titles[i].text = ""
                 }
             }
+
         }
 
-        private fun checkViewsWidth() {
+        private fun checkViewsWidth(itemView: View) {
```

Comment:
Let's add some docs.
```suggestion
        /**
        * Bla bla bla
        *
        */
        private fun checkViewsWidth(itemView: View) {
```